### PR TITLE
build: adjust android build path

### PR DESCRIPTION
**Pull request by compnerd**
- Remove custom `SWIFT_ANDROID_NDK_PATH` variable for the CMake variable `CMAKE_ANDROID_NDK` which is part of the default CMake tooling.
- Adjust the default NDK sysroot to be the NDK sysroot, not the termux sysroot to match the SDK name.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @etcwilde @finagolfin

**Comment by compnerd** [comment]
CC: @drodriguez

**Review by drodriguez — COMMENTED** [review]
I am +1 in changing to a variable that CMake understands and might be used for other things. However a simple grepping of the variable name can find a lot of cases that are not dealt with here (the `build-script-impl`, the test files, CMake files for libdispatch).

BTW, the Android CI has been broken (and it cannot be launched on demand from a PR) since `main` required using Swift Syntax for some pieces of the toolchain.

**Comment by compnerd** [comment]
I can likely go through the tests; I noticed the libdispatch issue locally and need to dig into that. I am running this locally for now. I will be putting up some changes subsequently to start building the Android SDKs on Windows to ensure that we don't regress those builds.

**Comment by etcwilde** [comment]
Yeah, separating termux and Android makes sense since they are different userspaces.

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

File: CMakeLists.txt [hunk 1]
```diff
@@ -1094,8 +1092,9 @@ elseif("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "ANDROID")
   set(SWIFT_HOST_VARIANT "android" CACHE STRING
       "Deployment OS for Swift host tools (the compiler) [android]")
 
-  set(SWIFT_ANDROID_NATIVE_SYSROOT "/data/data/com.termux/files" CACHE STRING
-      "Path to Android sysroot, default initialized to the Termux app's layout")
```
Comment (finagolfin) [inline CMakeLists.txt:1098]: You misunderstand what this is used for. `SWIFT_ANDROID_NATIVE_SYSROOT` is a native sysroot used in Android itself, eg the Termux app that runs on Android has its own native compilation sysroot and does not use the NDK. `SWIFT_ANDROID_NDK_PATH` is the path to the NDK and is _only_ set when cross-compiling to Android.

That's why the code in `cmake/modules/SwiftAndroidSupport.cmake` below [first checks if `SWIFT_ANDROID_NDK_PATH` is set](https://github.com/apple/swift/pull/71927/files#diff-9b6a95e185bd44a7c5bcb1e998923efe6ac51b1f9c4dd072e0dde723dd88b19cL2), so it knows if it is cross-compiling with the NDK, then [checks if natively compiling with an Android sysroot, ie in Termux](https://github.com/apple/swift/pull/71927/files#diff-9b6a95e185bd44a7c5bcb1e998923efe6ac51b1f9c4dd072e0dde723dd88b19cR5), and if neither happens to be set, it errors.

Don't modify these two lines: it is not relevant to the normal Android cross-compilation workflow that you are using.

**Comment by finagolfin** [comment]
> Yeah, separating termux and Android makes sense since they are different userspaces.

They are already completely separated since [I added this `SWIFT_ANDROID_NATIVE_SYSROOT` variable 5 years ago](https://github.com/apple/swift/commit/796d6ade9ae9#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR751).

`SWIFT_ANDROID_NDK_PATH` is used when cross-compiling with the NDK and predates my involvement with Swift. `SWIFT_ANDROID_NATIVE_SYSROOT` is used when natively building in Termux on Android itself, which does not use the NDK and I presume you all are not doing, so it should not be touched by this pull.

**Comment by finagolfin** [comment]
I am fine with the medium-term goal of pulling in CMake's built-in support for Android cross-compilation- provided the unrelated changes to ` SWIFT_ANDROID_NATIVE_SYSROOT`, which is only used for natively compiling _on_ an Android device, are removed from this pull- but we currently do not use that built-in support and roll our own cross-compilation for _every_ platform, so without further changes, `CMAKE_ANDROID_NDK` will not be set. If you make those additional changes to define it, I suspect they will have to be fairly extensive in the CMake build config.

> the Android CI has been broken (and it cannot be launched on demand from a PR) since main required using Swift Syntax for some pieces of the toolchain.

That is not required yet but yours is the only CI disabling it, so every time some issue like #71807 is fixed, new issues crop up, since all the official CI enable using the new swift-syntax parser.

**Comment by finagolfin** [comment]
What were you building, were you cross-compiling the Swift compiler itself from Windows to Android? Because [unless the `SWIFT_HOST_VARIANT_SDK` is `ANDROID`, `SWIFT_ANDROID_NATIVE_SYSROOT` is not set](https://github.com/apple/swift/blob/f292ec9784efd5d6bdb8b7c5f1c11e1a534709f9/CMakeLists.txt#L1092).

> we should rename it to reflect that it is for termux and not Android

It is for _native_ compilation- not Termux, as there are other native compilation environments on Android- but simply default-initialized to Termux, as Termux is by far the most popular app used for that. The name `SWIFT_ANDROID_NATIVE_SYSROOT` was chosen to reflect native compilation on Android, feel free to rename it in this pull if you have a better name in mind.

**Comment by compnerd** [comment]
> What were you building, were you cross-compiling the Swift compiler itself from Windows to Android?

I'm building the runtime only not the compiler. I am cross-compiling from Windows to Android though.

**Pull request closed without merging**